### PR TITLE
feat: 🎸 Add See More Section

### PR DESCRIPTION
**Pull request by Malayvasa**
Related to #384

### CODE REVIEW

File: src/components/DataRoom/SeeMore/Wave.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    const updateWidth = (entries: ResizeObserverEntry[]) => {
+      setWidth(entries[0].contentRect.width)
+    }
+
+    const resizeObserver = new ResizeObserver(updateWidth)
+    resizeObserver.observe(containerRef.current)
+
+    return () => resizeObserver.disconnect()
+  }, [])
```
Comment (iamacook) [inline src/components/DataRoom/SeeMore/Wave.tsx:43]: Nit: let's move this to a custom hook. We could make it reusable and return the width _and_ height.

File: src/components/DataRoom/SeeMore/Wave.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      const elapsed = time - startTimeRef.current
+
+      const path = Array.from({ length: width + 1 }, (_, i) => {
+        const y = height / 2 + amplitude * Math.sin((i / width) * 2 * Math.PI * frequency + (elapsed * speed) / 1000)
+        return `${i},${y}`
+      }).join(' L')
+
+      pathRef.current?.setAttribute('d', `M${path}`)
+
+      requestRef.current = requestAnimationFrame(animate)
```
Comment (iamacook) [inline src/components/DataRoom/SeeMore/Wave.tsx:57]: We don't need to use a ref for this. We can use a local variable.

File: src/components/DataRoom/SeeMore/index.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  const backgroundRef = useRef<HTMLDivElement>(null)
+  const isMobile = useIsMediumScreen()
+
+  const { scrollYProgress } = useScroll({
+    target: backgroundRef,
+    offset: ['start end', 'end start'],
+  })
+
+  const opacity = useTransform(scrollYProgress, [0, 0.3, 0.6, 0.75], [0, 1, 1, 0])
+  const opacitText = useTransform(scrollYProgress, [0, 0.3, 0.5, 0.75], [0, 1, 1, 0])
```
Comment (DiogoSoaress) [inline src/components/DataRoom/SeeMore/index.tsx:19]:
```suggestion
  const textOpacity = useTransform(scrollYProgress, [0, 0.3, 0.5, 0.75], [0, 1, 1, 0])
```

File: src/components/DataRoom/SeeMore/styles.module.css [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+.wave3 {
+  position: absolute;
+  top: 50%;
+  width: 100%;
+  transform: translateY(-50%);
+  right: 0;
+  z-index: 10;
+}
```
Comment (DiogoSoaress) [inline src/components/DataRoom/SeeMore/styles.module.css:60]: Can't this be the same class?

File: src/components/DataRoom/SeeMore/styles.module.css [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    transform: translateY(-25%);
+  }
+
+  .wave2 {
+    transform: translateY(-25%);
+  }
+
+  .wave3 {
+    transform: translateY(-25%);
+  }
```
Comment (DiogoSoaress) [inline src/components/DataRoom/SeeMore/styles.module.css:82]: Same class

File: src/hooks/useContainerSize.ts
Comment (DiogoSoaress) [inline src/hooks/useContainerSize.ts:1]: Isn't this hook achieving the same as your other PR? https://github.com/safe-global/safe-homepage/pull/427/files#diff-e2177b83efbc74640ebcdbe372bcab4c7a8ddc7bc30f8b3dfce99761fdb54ad0

File: src/components/DataRoom/SeeMore/Wave.tsx [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+      const elapsed = time - startTime
+
+      const path = Array.from({ length: width + 1 }, (_, i) => {
+        const y = height / 2 + amplitude * Math.sin((i / width) * 2 * Math.PI * frequency + (elapsed * speed) / 1000)
+        return `${i},${y}`
+      }).join(' L')
+
+      const pathElement = containerRef.current?.querySelector('path')
+      if (pathElement) {
+        pathElement.setAttribute('d', `M${path}`)
```
Comment (DiogoSoaress) [inline src/components/DataRoom/SeeMore/Wave.tsx:44]: I am having a runtime error coming from this line
Screenshot 2024-08-12 at 12 22 25

**Review by DiogoSoaress — APPROVED** [review]
Good job! There is one last comment but it should be good to merge after

**Review by DiogoSoaress — CHANGES_REQUESTED** [review]
I was also observing huge CPU usage in this branch and in the `Wave.tsx` file the sin computation and rendering is running as fast as possible the whole time…. We should try to only compute once the sin waves values or load them from a JSON file

**Review by DiogoSoaress — CHANGES_REQUESTED** [review]
Some leftovers

**Review by DiogoSoaress — APPROVED** [review]

**Pull request merged**